src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -36,6 +36,11 @@ public class WebScriptHostManager : ScriptHostManager
         private bool _warmupComplete = false;
         private bool _hostStarted = false;
 
+        // for mock testing
+        public WebScriptHostManager()
```

Comment:
Could we pass mocked dependencies into the constructor below to avoid having this constructor? Having this public constructor could be confusing. Removing it would minimize the chances of having this this type initialized in a *bad* state. (if we could at least make this `internal` calling the base with what it needs, that would help)